### DISCUSSION TITLE
Install or update appium-gestures-plugin required for W3c Actions/Gestures implementation when using Appium 2.0

### DISCUSSION THREAD

**Discussion by vnuta9** [appium/appium#18715] [Q&A] [answered]
I am using Appium2.0.0-beta66 version. Since TouchAction class being used for the gesture/actions is being deprecated, I am supposed to install appium-gestures-plugin ([from](https://github.com/AppiumTestDistribution/appium-gestures-plugin) ) and use it.

When I try to run the command

`$appium plugin install --source=npm appium-gestures-plugin`
as was suggested, I am getting the message

✔ Installing 'appium-gestures-plugin'
Error: ✖ A plugin named "gestures" is already installed. Did you mean to update?
I tried to check the installed plugins

`$appium plugin list --installed`
✔ Listing installed plugins

- relaxed-caps@1.0.0-beta.13 [installed (npm)]
- images@1.3.7 [installed (npm)]
- execute-driver@2.0.6 [installed (npm)]
- gestures@1.0.0-beta.4 [installed (npm)]

Then ran this command:
`$appium plugin update  appium-gestures-plugin
`
I got back this message:
**Error: ✖ The plugin "appium-gestures-plugin" was not installed, so can't be updated**

**How to update the existing 'gestures@1.0.0-beta.4' to gestures@2.0.0 ?**

**Comment by jlipps** [comment]
Answered here https://github.com/AppiumTestDistribution/appium-gestures-plugin/issues/74

**Reply by vnuta9** [reply]
@jlipps
Thanks for your helpful reply.
When I ran the command,
$appium plugin update gestures
I got the message:
✔ Checking if plugin 'gestures' is updatable
✔ Checking if plugin 'gestures' needs an update
ℹ Update report:
✖ - 'gestures' failed to update: Error: ✖ The plugin 'gestures' has a major revision update (1.0.0-beta.4 => 2.0.0), which could include breaking changes. If you want to apply this update,

re-run with --unsafe

Is it OK to run that command ??

appium plugin update gestures --unsafe

I am assuming that it is only suggesting that it is updating to ver2.0, which is not backward compatible, plus it is same as the 'appium-gestures-plugin', even though the name differs.

**Reply by vnuta9** [reply]
@jlipps @saikrishna321 
Hello Gurus;
How do I incorporate 'gesture' and 'wait' plugins in to the automatic/programmatic starting of the appium server ? Presently, I am using this code...
```
    public AppiumDriverLocalService **GetAppiumService**() {
        GlobalParams params = new GlobalParams();
        return AppiumDriverLocalService.buildService(new AppiumServiceBuilder()
                .usingAnyFreePort()
                .withTimeout(Duration.ofSeconds(40)) // based on Mykola Mokhnac suggestion to stop error msg
                .withArgument(() -> "--base-path", "/wd/hub")
                .withArgument(GeneralServerFlag.SESSION_OVERRIDE)
                .withLogFile(new File(params.getPlatformName() + "_"
                        + params.getDeviceName() + File.separator + "Server.log")));
    }
```

**Reply by SrinivasanTarget** [reply, marked as answer]
You can start activating plugin using:

withArgument(GeneralServerFlag.USE_PLUGINS, "gestures,element-wait"));

**Reply by vnuta9** [reply]
@SrinivasanTarget 
Thanks for helpful reply. 
Now the way to do if I have to use two plugins is 
```
    public AppiumDriverLocalService **GetAppiumService**() {
        GlobalParams params = new GlobalParams();
        return AppiumDriverLocalService.buildService(new AppiumServiceBuilder()
                .usingAnyFreePort()
                .withTimeout(Duration.ofSeconds(40)) // based on Mykola Mokhnac suggestion to stop error msg
                .withArgument(GeneralServerFlag.USE_PLUGINS, "gestures") // <<<-------
                .withArgument(GeneralServerFlag.USE_PLUGINS, "element-wait")  // <<<-------
                .withArgument(() -> "--base-path", "/wd/hub")
                .withArgument(GeneralServerFlag.SESSION_OVERRIDE)
                .withLogFile(new File(params.getPlatformName() + "_"
                        + params.getDeviceName() + File.separator + "Server.log")));
    }
```
Thanks a lot,

**Reply by SrinivasanTarget** [reply]
Just give all the plugins you want to activate as comma separated values.

**Comment by TheSRK** [comment]
GeneralServerFlag.USE_PLUGINS has been deprecated, what should we use now? Please help @SrinivasanTarget !! 
I want to use wait plugin while opening appium served programmatically.

**Reply by vnuta9** [reply]
I was using as below...
 .withArgument(GeneralServerFlag.USE_PLUGINS, "gestures,element-wait") // <<<------- or comma separated list

What is the updated method of doing the same in the light of the deprecation as stated by @TheSRK
